### DISCUSSION TITLE
1.2.0 ubuntu x64: problem with nvidia graphic card

### DISCUSSION THREAD

**Discussion by trekfan1** [rustdesk/rustdesk#2523] [General]
If I start a connection session from a PC with a NVIDIA graphics card, the window closes almost immediately, if instead I connect to the same PC remotely no problem.

**Comment by rustdesk** [comment]
nightly build?

**Reply by trekfan1** [reply]
last nightly build

**Comment by 21pages** [comment]
disable hardware codec and try.
![1670845622411](https://user-images.githubusercontent.com/14891774/207037678-d67012e5-3c08-4ec9-9326-2fe9e45dc8f5.png)

Rename available.jpg below as available.out and run in terminal.
![available](https://user-images.githubusercontent.com/14891774/207043969-efc6b9cc-fb53-4109-8492-4c3a1d938a9d.jpg)

 Enable hardware codec and retry by `sudo apt-get install libva-dev libvdpau-dev` if these two libs are not installed.

**Reply by rustdesk** [reply]
It is Ubuntu

**Reply by 21pages** [reply]
update

**Comment by trekfan1** [comment]
The recommended files installed and disabled the hardware codec but nothing, the same result, starting from the terminal after the crash appears a long string of codes, of which I only put the final part

`` nouveau: 0x200205f7
Nouveau: 0x00000000
Nouveau: 0x00000054
Nouveau: 0x80000585
Nouveau: 0x200406c0
Nouveau: 0x00000000
Nouveau: 0x00215000
Nouveau: 0x000000d3
Nouveau: 0x1000f010
rustesk: ../nouveau/pushbuf.c:730: nouveau_pushbuf_data: "kref" assertion failed .`

**Reply by xalt7x** [reply]
@trekfan1 
> The recommended files installed and disabled the hardware codec but nothing, the same result, starting from the terminal after the crash appears a long string of codes, of which I only put the final part
> 
> `` nouveau`

Did you try to replace built-in Nouveau Linux kernel driver with Nvidia proprietary driver?
Most distros should have packages for that. Some of them (PopOS, Ubuntu, Linux Mint, Manjaro) even have GUI tools for simple installations.

**Comment by 21pages** [comment]
1. Does  `/root/.config/rustdesk/RustDesk_hwcodec.toml` exist on your pc? It contains the codec info used on your pc, you can paste it out If it's exist.
2. Did available.out crash? you can paste log as jpg file if it's too long.
3. Try [this](https://github.com/21pages/rustdesk/releases/tag/nightly) or tomorrow nightly,disable hardware codec option, it should not crash.

**Comment by trekfan1** [comment]
`[options]
bestHwDecoders = '''
{
  "h264": {
    "name": "h264",
    "format": "H264",
    "vendor": "OTHER",
    "score": 80,
    "hwdevice": "AV_HWDEVICE_TYPE_NONE"
  },
  "h265": {
    "name": "hevc",
    "format": "H265",
    "vendor": "OTHER",
    "score": 80,
    "hwdevice": "AV_HWDEVICE_TYPE_NONE"
  }
}'''
bestHwEncoders = '''
{
  "h264": null,
  "h265": null
}'''
`

**Reply by 21pages** [reply]
Only soft h264/h265 decoders.

**Reply by rustdesk** [reply]
it seems https://github.com/MultiMC/Launcher/issues/3003 nvidia driver not supported on some Ubuntu. Can we test first if support before turning on?

**Reply by trekfan1** [reply]
The graphic Card is a old GT610 (it's not my pc but a Parish PC), i have tried to install the last nightly build but same result

**Reply by trekfan1** [reply]
Same pc but with windows no problem

**Reply by trekfan1** [reply]
I m tried on another pc with a GEFORCE GT1030 but same result on ubuntu

**Reply by 21pages** [reply]
> same result

close immediately or not work? check driver by `lsmod | grep nvidia`

**Reply by trekfan1** [reply]
Sometimes it ends immediately, sometimes after soon as soon as the mouse moves, here is the report:

lsmod
Module                  Size  Used by
btrfs                1413120  0
blake2b_generic        20480  0
xor                    24576  1 btrfs
raid6_pq              122880  1 btrfs
ufs                    90112  0
qnx4                   16384  0
hfsplus               118784  0
hfs                    73728  0
minix                  49152  0
ntfs                  114688  0
msdos                  20480  0
jfs                   212992  0
xfs                   946176  0
libcrc32c              16384  2 btrfs,xfs
snd_seq_dummy          16384  0
snd_hrtimer            16384  1
nouveau              2342912  23
snd_hda_codec_realtek   167936  1
snd_hda_codec_generic    98304  1 snd_hda_codec_realtek
snd_hda_codec_hdmi     73728  1
ledtrig_audio          16384  1 snd_hda_codec_generic
edac_mce_amd           36864  0
snd_hda_intel          45056  4
snd_intel_dspcfg       36864  1 snd_hda_intel
snd_intel_sdw_acpi     20480  1 snd_intel_dspcfg
snd_hda_codec         167936  4 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_realtek
kvm_amd               143360  0
ccp                   110592  1 kvm_amd
snd_hda_core          102400  5 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek
snd_seq_midi           20480  0
snd_seq_midi_event     16384  1 snd_seq_midi
snd_hwdep              16384  1 snd_hda_codec
kvm                   716800  1 kvm_amd
snd_pcm               151552  5 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,snd_hda_core
input_leds             16384  0
mxm_wmi                16384  1 nouveau
irqbypass              16384  1 kvm
snd_rawmidi            49152  1 snd_seq_midi
video                  61440  1 nouveau
drm_ttm_helper         16384  1 nouveau
ttm                    86016  2 drm_ttm_helper,nouveau
wmi_bmof               16384  0
serio_raw              20480  0
k10temp                16384  0
snd_seq                86016  9 snd_seq_midi,snd_seq_midi_event,snd_seq_dummy
sp5100_tco             20480  0
drm_display_helper    176128  1 nouveau
binfmt_misc            24576  1
cec                    77824  1 drm_display_helper
rc_core                65536  1 cec
snd_seq_device         16384  3 snd_seq,snd_seq_midi,snd_rawmidi
snd_timer              40960  3 snd_seq,snd_hrtimer,snd_pcm
drm_kms_helper        204800  2 drm_display_helper,nouveau
snd                   114688  20 snd_hda_codec_generic,snd_seq,snd_seq_device,snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek,snd_timer,snd_pcm,snd_rawmidi
i2c_algo_bit           16384  1 nouveau
fb_sys_fops            16384  1 drm_kms_helper
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
soundcore              16384  1 snd
mac_hid                16384  0
msr                    16384  0
parport_pc             49152  0
ppdev                  24576  0
lp                     20480  0
parport                73728  3 parport_pc,lp,ppdev
drm                   581632  13 drm_kms_helper,drm_display_helper,drm_ttm_helper,ttm,nouveau
ramoops                32768  0
reed_solomon           24576  1 ramoops
pstore_blk             16384  0
pstore_zone            28672  1 pstore_blk
efi_pstore             16384  0
ip_tables              32768  0
x_tables               57344  1 ip_tables
autofs4                49152  2
hid_generic            16384  0
uas                    32768  0
r8169                  90112  0
usbhid                 65536  0
realtek                32768  1
usb_storage            81920  1 uas
hid                   155648  2 usbhid,hid_generic
psmouse               180224  0
mdio_devres            16384  1 r8169
libphy                172032  3 r8169,mdio_devres,realtek
i2c_piix4              28672  0
ahci                   49152  2
libahci                49152  1 ahci
wmi                    36864  4 video,wmi_bmof,mxm_wmi,nouveau

**Reply by 21pages** [reply]
#2543 , maybe render problem, try `LIBGL_ALWAYS_SOFTWARE=true ./rustdesk`, and you didn't install nvidia official driver.

**Reply by 21pages** [reply]
Could you help trying  `LIBGL_ALWAYS_SOFTWARE=true ./rustdesk`?

**Reply by trekfan1** [reply]
It Works! However, this graphic card does not need official (owners) Nvidia drivers, as if you install them, the screen resolution drops and I can't change it. Nouveau must be used.

Here is the output of the terminal:

Flutter: launch args:
Flutter: FFI initialization Main
Flutter: _Apptype: Main, info1-ID: 7b022ad462534Fe58B9af42edbe4f2c, info2-name: ubuntu, dir:/home/sacristy/docume
Flutter: _globalffi Init
Flutter: _globalffi Init End
Flutter: RegistavenThandler callback_quary_online peer recent
Flutter: RegistavenThandler Load_recent_peers Recently Peer
Flutter: [MultiwindowHandler] Active window modified: [0]
Flutter: [MultiwindowHandler] Active window modified: [0, 1]
On the modification of the State: 0x2378520on Sample State: 0x2378520on Sample State: 0x2378520ON State Edit: 0x2378520ON State Edit: 0x2378520flutter: Lancia Args: [Multi_Window, 1, {"Type": 1, "ID": "435471873"}]
Flutter: Remote FFI initialization
Flutter: _Apptype: Remote, Info1-ID: 7B022AD462534FE58B9AF42EDBE4F2C, info2-name: ubuntu, dir:/home/sacristy/documes
Flutter: _globalffi Init
Flutter: _globalffi Init End
Flutter: theme theme Registerventhandler
Flutter: language of the language registinenthandler
Flutter: [MAIN] Call Get_Window_info with Args from the window 1
Flutter: Onfocuschange (active window: true) True
Flutter: Restore LPOS: 1280.0/720.0, Offset: 0.0/27.0
Flutter: [MAIN] Call Get_Window_info with Args from the window 1
Flutter: Onfocuschange (active window: True) False
Flutter: False Ismaximized
Flutter: {Restart: False}
Flutter: Onfocuschange (active window: true) True
Flutter: Onfocuschange (active window: True) False
Flutter: [MAIN] Call Get_Window_info with Args from the window 1
Flutter: Onfocuschange (active window: true) True
Flutter: [MAIN] Call Get_Window_info with Args from the window 1
Flutter: Personalized cursor registers with key 435471873_65539_320000000_320000000
Flutter: Onfocuschange (active window: True) False
On the modification of the State: 0x2378520on Modification of the State: 0x2378520flutter: Onfocuschange (active window: True) True
Flutter: Onpointdownimage
Flutter: remote page enamel 435471873
Flutter: Elimination of the cursor with the key 435471873_65539_320000000_320000000
Flutter: closed model
On the modification of the state: 0x2378520flutter: [MAIN] Call Hide WITH ARGS {ID: 1} From the window 1
Flutter: [MultiwindowHandler] Active window modified: [0]
On the modification of the state: 0x2378520

Hope my english is good, i 'm italian and the initial text before the output is translated by google translate

**Reply by 21pages** [reply]
Thanks, it's really helpful!

**Reply by 21pages** [reply]
When it crashes for this reason, I want to give advice to the user to install  nouveau, and after that run with env LIBGL_ALWAYS_SOFTWARE  automatically. So I want to specify the crash backtrace to know it is the crash caused by this reason. l log the backtrace to `/tmp/log.txt`,  Could you download and run [this](https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.0-x86_64-unknown-linux-gnu-ubuntu-18.04.deb) without env LIBGL_ALWAYS_SOFTWARE? I need to see the backtrace log, and it may not crash because the signal is caught.

**Reply by trekfan1** [reply]
Done
[log.txt](https://github.com/rustdesk/rustdesk/files/10341768/log.txt)

**Reply by 21pages** [reply]
Thanks! Did it crash in this version?
Not all, but most contains `nouveau_pushbuf_kick`. I think it's the key function of nouveau driver user

**Reply by trekfan1** [reply]
It's crashed, but after that i have installed the package nvidia-vaapi-driver and rustdesk seems working but not perfectly, i suggest to add this package as dependency

**Reply by 21pages** [reply]
Good clue, what does `not perfectly` mean?

**Reply by trekfan1** [reply]
When I connect to a PC locally, it connects but afterwards it seems to stop for a while as if the PC was freezed but it is not, I will do further tests as soon as I can

**Reply by 21pages** [reply]
Can you please try this one? https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.0-x86_64-unknown-linux-gnu-ubuntu-18.04-fix-nvidia-flutter.deb

It will find out from the crash stack whether software rendering is required, and then set the options for that. After restarting the main window, it will automically set `LIBGL_ALWAYS_SOFTWARE=true` for rustdesk process.

After crash, the option should have been set.
![1672905606802](https://user-images.githubusercontent.com/14891774/210730621-39c2b676-8a49-4711-bcfd-01c7f5b4dd25.png)

The crash stack could be seen at `$HOME/.local/share/logs/RustDesk/rustdesk_rCURRENT.log`

**Reply by trekfan1** [reply]
[rustdesk_rCURRENT.log](https://github.com/rustdesk/rustdesk/files/10357528/rustdesk_rCURRENT.log)

**Reply by 21pages** [reply]
Did it exit? If it exit,the log should be in the most recent dated file in directory `$HOME/.local/share/logs/RustDesk`

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/2543

**Reply by trekfan1** [reply]
The point is that sometimes it works and sometimes crashes, I don't understand why. This is the log after it seems to have worked
[rustdesk_rCURRENT.log](https://github.com/rustdesk/rustdesk/files/10360495/rustdesk_rCURRENT.log)

This is the log when it is crashed instead

[rustdesk_rCURRENT (copia).log](https://github.com/rustdesk/rustdesk/files/10360503/rustdesk_rCURRENT.copy.log)

**Reply by 21pages** [reply]
Thanks for your test. Could you see `Got signal 11 and exit` in the log file? And the call stack is below that line. [rustdesk_rCURRENT (copia).log](https://github.com/rustdesk/rustdesk/files/10360503/rustdesk_rCURRENT.copy.log) was broken. Have the option `Always use software rendering` been set automatically? If not, what about set it manually?

**Comment by 21pages** [comment]
If someone uses nvidia official graphics card and this situation happens, it will be very helpful if you can run [https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.0-x86_64-unknown-linux-gnu-ubuntu-18.04-nvidia-flutter-log.deb](https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.0-x86_64-unknown-linux-gnu-ubuntu-18.04-nvidia-flutter-log.deb) and it will print the log to /tmp/log.txt.

**Reply by Stayer** [reply]
this package is working good. But latest build of rustdesk gives same problem

**Reply by trekfan1** [reply]
I have tested on my notebook with a GeForce 930MX: on linux same problem, on windows all ok
